modules/ui/fields/date.js

```diff
@@ -0,0 +1,268 @@
+import { dispatch as d3_dispatch } from 'd3-dispatch';
+import { select as d3_select } from 'd3-selection';
+import * as countryCoder from '@ideditor/country-coder';
+
+import { uiCombobox } from '../combobox';
+import { t, localizer } from '../../core/localizer';
+import { utilGetSetValue, utilNoAuto, utilRebind, utilTotalExtent } from '../../util';
+
+
+export function uiFieldDate(field, context) {
+    let dispatch = d3_dispatch('change');
+    let yearInput = d3_select(null);
+    let eraInput = d3_select(null);
+    let monthInput = d3_select(null);
+    let dayInput = d3_select(null);
+    let _entityIDs = [];
+    let _tags;
+
+    let dateTimeFormat = new Intl.DateTimeFormat(localizer.languageCode(), {
+        year: 'numeric',
+        era: 'short',
+        month: 'long',
+        day: 'numeric',
+        timeZone: 'UTC',
+    });
+    let formatParts = dateTimeFormat.formatToParts(new Date()).map(p => p.type);
+
+    /**
+     * Returns the localized name of the era in the given format.
+     *
+     * @param year A representative year within the era.
+     */
+    function getEraName(year, format) {
+        let longFormat = new Intl.DateTimeFormat(localizer.languageCode(), {
+            year: 'numeric',
+            era: format,
+            timeZone: 'UTC',
+        });
+        let eraDate = new Date(Date.UTC(0, 0, 1));
+        eraDate.setUTCFullYear(year);
+        let parts = longFormat.formatToParts(eraDate);
+        let eraPart = parts.find(p => p.type === 'era');
+        return eraPart && eraPart.value;
+    }
+
+    let bceName = getEraName(0, 'short');
+    let ceName = getEraName(1, 'short');
+    let eraNames = [
+        {
+            id: 'bce',
+            value: bceName,
+            title: bceName,
+            display: selection => selection.append('span')
+                .attr('class', 'localized-text')
+                .text(bceName ? bceName : 'BCE'),
+            terms: [getEraName(0, 'long'), getEraName(0, 'narrow')],
+        },
+        {
+            id: 'ce',
+            value: ceName,
+            title: ceName,
+            display: selection => selection.append('span')
+                .attr('class', 'localized-text')
+                .text(ceName ? ceName : 'CE'),
+            terms: [getEraName(1, 'long'), getEraName(1, 'narrow')],
+        },
+    ];
+    let eraCombo = uiCombobox(context, 'date-era')
+        .data(eraNames);
+
+    /// Returns the localized name of a month in the given format.
+    function getMonthName(month, format) {
+        let longFormat = new Intl.DateTimeFormat(localizer.languageCode(), {
+            month: format,
+            timeZone: 'UTC',
+        });
+        let parts = longFormat.formatToParts(new Date(Date.UTC(0, month, 1)));
+        let monthPart = parts.find(p => p.type === 'month');
+        return monthPart && monthPart.value;
+    }
+
+    let monthNames = Array.from({length: 12}, (_, i) => getMonthName(i, 'long'))
```

Comment:
The long month format isn’t great for Chinese. It winds up spelling out the month number.